src/main/java/org/folio/persist/InstanceRepository.java

```diff
@@ -40,6 +40,24 @@ public InstanceRepository(Context context, Map<String, String> okapiHeaders) {
     super(postgresClient(context, okapiHeaders), INSTANCE_TABLE, Instance.class);
   }
 
+  public Future<Boolean> doesInstanceExistBySubjectField(String field, String sourceId) {
+    var sql = new StringBuilder("SELECT exists ( ");
+    sql.append("SELECT 1 FROM ");
+    sql.append(INSTANCE_TABLE);
+    sql.append(String.format(" WHERE jsonb @> '{\"subjects\": [{\"%s\": \"%s\"}] }' ", field, sourceId));
+    sql.append(") as record_exists");
+
+    return postgresClient.execute(sql.toString())
+      .map(rowSet -> {
+        if (rowSet != null && rowSet.iterator().hasNext()) {
+          Row row = rowSet.iterator().next();
+          return row.getBoolean("record_exists");
+        }
+        return Boolean.FALSE;
+      });
+  }
```

Comment:
This approach could be highly inefficient; in the worst case, it would require searching through all records to find one with the subject field, potentially impacting other processes related to instances and subject types.

I suggest creating an intermediate many-to-many relationship table with foreign keys (FKs) linked to both subject types and instances. This would facilitate reliance on database validation and eliminate the need to search the instance table.